### PR TITLE
PB-1884: Use the new stork Snapshotter interface in DataExport reconcilor

### PR DESCRIPTION
**Pull request by adityadani**
**What this PR does / why we need it**:
-  Minor modifications to use the new interface.
- Do the pre checks in the reconcilor for kopia backup and restore.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:
Work in Progress PR. Still facing issues triggering a snapshot and performing a restore as a part of the DataExport CR.

### CODE REVIEW

File: pkg/controllers/dataexport/dataexport.go [hunk 1]
```diff
@@ -27,13 +28,15 @@ var (
...
+	client      runtimeclient.Client
+	snapshotter snapshotter.Snapshotter
 }
 
 // NewController returns a new instance of the controller.
 func NewController(mgr manager.Manager) (*Controller, error) {
 	return &Controller{
-		client: mgr.GetClient(),
+		client:      mgr.GetClient(),
+		snapshotter: snapshotter.NewDefaultSnapshotter(),
```
Comment (ram-infrac) [inline pkg/controllers/dataexport/dataexport.go:39]: nice

File: pkg/controllers/dataexport/reconcile.go [hunk 2]
```diff
@@ -291,47 +302,85 @@ func (c *Controller) stageSnapshotScheduled(ctx context.Context, snapshotter sna
...
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", err
+	}
+
+	cs, err := kSnapshotClient.NewForConfig(config)
+	if err != nil {
+		return "", err
+	}
+	_, err = cs.SnapshotV1beta1().VolumeSnapshotClasses().Get(context.TODO(), dataExport.Spec.SnapshotStorageClass, metav1.GetOptions{})
```
Comment (ram-infrac) [inline pkg/controllers/dataexport/reconcile.go:319]: aren't we going to decide this by looking at pvc?

**Review by ram-infrac — APPROVED** [review]
lgtm, after addressing comment

File: pkg/controllers/dataexport/reconcile.go (hunk 2 above)
Comment (adityadani) [reply pkg/controllers/dataexport/reconcile.go:329]: I thought about it, but ultimately the decision of what snapshot provider to use is really dependent on the snapshot class that was provided.

File: pkg/controllers/dataexport/reconcile.go [hunk 3]
```diff
@@ -25,18 +29,25 @@ import (
...
+	KopiaSecretName          = "generic-backup-repo"
+	KopiaSecretNamespace     = "kube-system"
+	backupCRNameKey          = "kdmp.portworx.com/backup-cr-name"
+	pvcNameKey               = "kdmp.portworx.com/pvc-name"
+	labelNamelimit           = 63
+	csiProvider              = "csi"
+	dataExportUIDAnnotation  = "portworx.io/dataexport-uid"
+	dataExportNameAnnotation = "portworx.io/dataexport-name"
+	skipResourceAnnotation   = "stork.libopenstorage.org/skip-resource"
+	// pvcNameLenLimit is the max lenght of PVC name that DataExport related CRs
```
Comment (prashanthpx) [inline pkg/controllers/dataexport/reconcile.go:48]: typo `lenght`

File: pkg/controllers/dataexport/reconcile.go [hunk 4]
```diff
@@ -276,21 +288,41 @@ func (c *Controller) stageInitial(ctx context.Context, snapshotter snapshots.Dri
...
+		return false, fmt.Errorf("failed to get snapshot driver for %v: %v", snapshotDriverName, err)
 	}
 
-	name, namespace, err := snapshotter.CreateSnapshot(
-		snapshots.PVCName(dataExport.Spec.Source.Name),
-		snapshots.PVCNamespace(dataExport.Spec.Source.Namespace),
-		snapshots.RestoreNamespaces(dataExport.Spec.Destination.Namespace),
+	// First check if a snapshot has already been triggered
+	snapName := toSnapName(dataExport.Spec.Source.Name, string(dataExport.UID))
+	/*_, err = snapshotDriver.SnapshotStatus(snapName, dataExport.Spec.Source.Namespace)
```
Comment (prashanthpx) [inline pkg/controllers/dataexport/reconcile.go:310]: Remove commented code.

**Review by prashanthpx — APPROVED** [review]
lgtm

**Pull request merged**